src/core/Tree/TreeNode.tsx

```diff
@@ -0,0 +1,257 @@
+      if (subNode.nodeId) {
+        nodes.push(subNode?.nodeId);
+      }
+    });
+    return nodes;
+  }, [subNodes]);
+
+  const styleLevel = React.useMemo(
+    () =>
+      getWindow()?.CSS?.supports?.(`--level: ${nodeDepth}`)
+        ? { '--level': nodeDepth, ...style }
+        : { marginLeft: nodeDepth ? nodeDepth * 28 : 0, ...style },
+    [nodeDepth, style],
+  );
+
+  const expandedNodes = React.useMemo(() => {
+    return context?.expandedNodes ?? [];
+  }, [context?.expandedNodes]);
+
+  const isExpanded = React.useMemo(() => {
+    return expandedNodes.findIndex((id) => id === nodeId) != -1;
+  }, [expandedNodes, nodeId]);
+
+  const onExpandChanged = (e: React.SyntheticEvent<Element, Event>) => {
+    if (isExpanded) {
+      context?.setExpandedNodes?.(
+        expandedNodes.filter((item) => item != nodeId),
+      );
+    } else {
+      context?.setExpandedNodes?.([...expandedNodes, nodeId]);
+    }
+
+    onNodeExpanded?.(nodeId, !isExpanded);
+    e.stopPropagation();
+  };
+
+  const visible = React.useMemo(() => {
+    let isParentExpanded = parentNode
+      ? expandedNodes.findIndex((id) => id === parentNode.data.nodeId) != -1
+      : true;
+    if (parentNode) {
+      let currentParent = parentNode.parent;
+      while (currentParent) {
+        if (
+          expandedNodes.findIndex((id) => id === currentParent?.data.nodeId) ===
+          -1
+        ) {
+          isParentExpanded = false;
+        }
+        currentParent = currentParent.parent;
+      }
+    }
+    return isParentExpanded;
+  }, [expandedNodes, parentNode]);
+
+  const [isChecked, setIsChecked] = React.useState(
+    React.isValidElement(nodeCheckbox) ? nodeCheckbox.props['checked'] : false,
+  );
+
+  const selectedNodes = React.useMemo(() => {
+    return context?.selectedNodes ?? [];
+  }, [context?.selectedNodes]);
+
+  const isSelected = React.useMemo(() => {
+    return selectedNodes.findIndex((id) => id === nodeId) != -1;
+  }, [selectedNodes, nodeId]);
+
+  const onNodeClick = () => {
+    if (isDisabled) {
+      return;
+    }
+    if (isSelected) {
+      context?.setSelectedNode?.(
+        selectedNodes.filter((item) => item != nodeId),
+      );
+    } else {
+      context?.setSelectedNode?.([nodeId]);
+    }
+    onNodeSelected?.(nodeId, !isSelected);
+  };
+
+  return (
+    <>
+      <li role='treeitem' id={nodeId} aria-expanded={isExpanded} {...rest}>
+        {visible && (
+          <div
+            className={cx(
+              'iui-tree-node',
+              {
+                'iui-active': isSelected,
+                'iui-disabled': isDisabled,
+              },
+              className,
+            )}
+            style={styleLevel}
+            onClick={() => onNodeClick()}
+            tabIndex={isSelected ? 0 : -1}
+          >
+            {nodeCheckbox && React.isValidElement(nodeCheckbox)
+              ? React.cloneElement(nodeCheckbox, {
+                  className: 'iui-tree-node-checkbox',
```

Comment:
I think we should only apply the class name and leave the rest to the user. Also, you need to merge `className` with its original one, check `IconButton how to do that.

Reply:
Fixed